### PR TITLE
Character counter

### PR DESCRIPTION
**Pull request by alexander-bobin**
https://www.pivotaltracker.com/story/show/97282192

- Create character counter and integrate into create petition flow
- Change JS unit testing from evergreen to jasmine-rails (which supports asset pipeline)
- Move all vendor JS to `/vendor/assets/javascripts/`
- Add `jquery-rails`

### CODE REVIEW

**Comment by davidbasalla** [comment]
some rebase action required

**Comment by alexander-bobin** [comment]
Sure thing. I'll rebase after we've got some feedback on the code or a LGTM

File: Gemfile.lock [hunk 1]
```diff
@@ -311,6 +307,3 @@ DEPENDENCIES
   webmock
   whenever
   will_paginate
-
-BUNDLED WITH
-   1.10.4
```
Comment (pixeltrix) [inline Gemfile.lock]: Leave this in please

File: spec/javascripts/character_counter_spec.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    $counter.remove();
+  });
+
+  describe("When called", function () {
+    it("Should put the max characters into the character counter", function () {
+      GOVUK.GDM.characterCounter();
+      expect($counter.text()).toEqual("50");
+    });
+
+    it("Should add an id to the character counter based on the textbox id", function () {
```
Comment (otlaitil) [inline spec/javascripts/character_counter_spec.js]: describing specs, don't use 'should'  - http://betterspecs.org/#should (is valid also with jasmine specs)

File: app/assets/javascripts/modules/character-counter.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        .html(charsRemaining)
+        .toggleClass('too-many-characters', charsRemaining < 0);
+    };
+
+  characterCounter = function () {
+    attach();
+  };
+
+  this.GOVUK = this.GOVUK || {};
+  this.GOVUK.GDM = this.GOVUK.GDM || {};
```
Comment (otlaitil) [inline app/assets/javascripts/modules/character-counter.js]: what does GDM stand for, Government Digital Marketplace? Do you think we should rename this?

**Comment by otlaitil** [comment]
Looks good! :+1:

**Comment by h-lame** [comment]
We might need to have a conversation about counting emoji properly.  That conversation probably needs to work across the server validation, db storage, and this JS counter, so I'm not sure this is the place to worry about it.  Just a thought I had when looking at the implementation (and influenced by recently reading [this epic commit/blog post](https://about.futurelearn.com/blog/a-commit-message-from-our-repo-2/))

File: app/assets/javascripts/modules/character-counter.js (hunk 3 above)
Comment (alexander-bobin) [inline app/assets/javascripts/modules/character-counter.js]: Good point. I'll change to PETS

**Comment by alexander-bobin** [comment]
Rebased. Good to go when green

**Pull request merged**